### PR TITLE
[Issue #2654] Next login flow backend

### LINKED ISSUE #2654: Task: Add login callback server action that receives simpler jwt and sets cookie for the frontend
### Summary

Add login callback server action that receives simpler jwt and sets cookie for the frontend.

### Acceptance criteria

- [ ] Route created
- [ ] Session cookie set for frontend that allows for frontend session

### PR DESCRIPTION
**Pull request by doug-s-nava**
## Summary
Fixes #2654

## Changes proposed

This is step 1 for implementing auth support on the Next JS application.

The change includes:
* a new route at /api/auth/callback. This is the route that will be hit by the flask API at the completion of the login flow on that side of the application. That request will include a JSON web token in a query param that will be parsed on the Next JS side and eventually set on the client as a cookie
* all necessary token parsing logic. Functions are created here for encrypting and decrypting the JWT and for getting and setting the token value from / on cookies
* support for a a "SESSION_SECRET" env var. This will be generated by terraform for each environment and used for encryption of the token before setting on the client
* an update to the terraform env vars doc, copy and pasted from the Nava platform
* a User page and corresponding error page to redirect to for temporary testing while we continue to build out the client side login implementation

### Test steps

1. start the app on this branch with `npm run dev`

#### New Login

1. hit http://localhost:3000/api/auth/callback?token=anyStringYouWant2Use with no cookies
2. _VERIFY_: page loads with `created session` message
3. use your chosen method of checking cookies on the page (ex. in Chrome, use the "Storage" tab of the developer tools), and look for a cookie named "session"
4. _VERIFY_: cookie is present (see screenshot)

#### Existing Login

1. hit http://localhost:3000/api/auth/callback with cookie set to a valid session cookie. Reach out to Doug if you need one, or you can grab the encrypted jwt set in the first test case
5. _VERIFY_: page loads with `already logged in` message
6. use your chosen method of checking cookies on the page (ex. in Chrome, use the "Storage" tab of the developer tools), and look for a cookie named "session"
4. _VERIFY_: cookie is present (see screenshot)

#### Error Cases

1. hit http://localhost:3000/api/auth/callback with no token param or cookies
2. _VERIFY_: user page loads with `no token provided` message
3. Note that there is a possible case of an "error!" message but I can't figure out a way to simulate it atm

### An example of a client side session cookie
Screenshot 2024-12-13 at 10 55 36 AM

### CODE REVIEW

File: frontend/src/app/[locale]/not-found.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default function NotFound() {
     <>
       <BetaAlert />
       <GridContainer className="padding-y-1 tablet:padding-y-3 desktop-lg:padding-y-15 measure-2">
-        <h1 className="nj-h1">{t("title")}</h1>
```
Comment (doug-s-nava) [inline frontend/src/app/[locale]/not-found.tsx:27]: looks like maybe a remnant from the nj ui project?

File: documentation/infra/environment-variables-and-secrets.md [hunk 2]
```diff
@@ -40,19 +40,23 @@ module "dev_config" {
...
-# environment-variables.tf
+# environment_variables.tf
 
 locals {
-  secrets = [
-    {
-      name           = "SOME_API_KEY"
-      ssm_param_name = "/${var.app_name}-${var.environment}/secret-sauce"
+  secrets = {
+    GENERATED_SECRET = {
```
Comment (mdragon) [inline documentation/infra/environment-variables-and-secrets.md:50]: Should this be more specific, like GENERATED_JWT_HASH_SECRET or GENERATED_JWT_SECRET?

**Review by mdragon — APPROVED** [review]
👍🏻

File: frontend/src/app/[locale]/not-found.tsx (hunk 1 above)
Comment (acouch) [reply frontend/src/app/[locale]/not-found.tsx:27]: Good catch.

File: frontend/src/app/[locale]/search/error.tsx [hunk 3]
```diff
@@ -13,12 +13,6 @@ import SearchBar from "src/components/search/SearchBar";
 import SearchFilters from "src/components/search/SearchFilters";
 import ServerErrorAlert from "src/components/ServerErrorAlert";
 
-interface ErrorProps {
```
[the remark this answers is not in the document]
Comment (acouch) [reply frontend/src/app/[locale]/search/error.tsx:16]: Maybe we need to look a little more at error handling across the app.

**Review by acouch — APPROVED** [review]
🚀

**Pull request merged**